### PR TITLE
Test on Travis-CI OS X

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by bluescarni** [comment]
@isuruf Thanks a lot! The builds look messy at the moment on all branches, I think that there were some changes on the travis infrastructures that broke things for piranha. It looks like the memory limit has been lowered, and there's a weird non-deterministic failure in one of the tests that I am trying to track down.

I'd like to close up the work in #87 and then we could rebase this branch on top of that, if that is ok with you. I'll go over the changes in this PR and have a few questions probably :)

Thanks a lot again!

**Comment by bluescarni** [comment]
@isuruf btw, so the changes to the ``thread_local`` test work ok on OSX? Should I ditch #90?

**Comment by isuruf** [comment]
@bluescarni, yes, the changes to thread_local works

**Comment by isuruf** [comment]
@bluescarni, if you want to get the rest of the dependencies on conda, ping me on gitter. It's a community effort called conda-forge, https://conda-forge.github.io.
I've already gotten piranha in, but pyranha is not.

**Comment by bluescarni** [comment]
@isuruf  Thanks a lot! If I understood correctly, doxygen and msgpack are the missing ones?

**Comment by isuruf** [comment]
Yes

File: src/kronecker_array.hpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ class kronecker_array
     static limit_type determine_limit(const size_type &m)
     {
         piranha_assert(m >= 1u);
-        std::mt19937 engine(static_cast<unsigned long>(m));
+        std::mt19937 engine(static_cast<unsigned>(m));
```
Comment (bluescarni) [inline src/kronecker_array.hpp]: @isuruf Good catch! Would you mind casting to ``std::mt19937::result_type`` instead?

File: tools/install_travis.sh [hunk 2]
```diff
@@ -5,30 +5,38 @@ set -e
...
-        cmake -DPIRANHA_WITH_MSGPACK=yes -DPIRANHA_WITH_BZIP2=yes -DPIRANHA_WITH_ZLIB=yes -DMSGPACK-C_INCLUDE_DIR=/home/travis/.local/include -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=yes -DPIRANHA_TEST_SPLIT=yes -DPIRANHA_TEST_SPLIT_NUM=${SPLIT_TEST_NUM} ../;
+        cmake -DPIRANHA_WITH_MSGPACK=yes -DPIRANHA_WITH_BZIP2=yes -DPIRANHA_WITH_ZLIB=yes -DCMAKE_PREFIX_PATH=$deps_dir -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=yes -DPIRANHA_TEST_SPLIT=yes -DPIRANHA_TEST_SPLIT_NUM=${SPLIT_TEST_NUM} ../;
         make VERBOSE=1;
         ctest -E "thread" -V;
     fi
 elif [[ "${BUILD_TYPE}" == "Coverage" ]]; then
-        cmake -DPIRANHA_WITH_MSGPACK=yes -DPIRANHA_WITH_BZIP2=yes -DPIRANHA_WITH_ZLIB=yes -DMSGPACK-C_INCLUDE_DIR=/home/travis/.local/include -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=yes -DCMAKE_CXX_FLAGS="-Og --coverage" -DPIRANHA_TEST_SPLIT=yes -DPIRANHA_TEST_SPLIT_NUM=${SPLIT_TEST_NUM} ../;
+        cmake -DPIRANHA_WITH_MSGPACK=yes -DPIRANHA_WITH_BZIP2=yes -DPIRANHA_WITH_ZLIB=yes -DCMAKE_PREFIX_PATH=$deps_dir -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=yes -DCMAKE_CXX_FLAGS="--coverage" -DPIRANHA_TEST_SPLIT=yes -DPIRANHA_TEST_SPLIT_NUM=${SPLIT_TEST_NUM} ../;
```
Comment (bluescarni) [inline tools/install_travis.sh]: In the last PR I introduced an additional flag, ``-Og`` here, in addition to ``--coverage``. This should take care of solving the travis timeouts while (hopefully) preserving the debug info. See for instance the timings for the coverage build here:

https://travis-ci.org/bluescarni/piranha/builds/175559999

Comment (isuruf) [inline tools/install_travis.sh]: Will fix that. This was a merge error.

File: src/kronecker_array.hpp (hunk 1 above)
Comment (isuruf) [inline src/kronecker_array.hpp]: Done.

**Comment by isuruf** [comment]
Tests pass. :smile:

**Comment by bluescarni** [comment]
@isuruf cheers!

**Pull request merged**